support/controlplane-component/defaults.go

```diff
@@ -51,6 +51,10 @@ func (c *controlPlaneWorkload) defaultOptions(cpContext ControlPlaneContext, pod
 		return nil, err
 	}
 
+	if c.serviceAccountKubeConfigOpts != nil {
+		c.addServiceAccountKubeconfigVolumes(podTemplateSpec)
```

Comment:
moved this before konnectivity-container injection to avoid adding a duplicate volumeMount, since konnectivity adds its own kubeconfig volumeMount.